### ISSUE TITLE
Can you share the implementation detail for baseline?(Cross-Entropy)

### ISSUE THREAD

**Issue by wjun0830** [alinlab/CSI#35]
Can you share how you implemented Cross-Entropy on Labeled ImageNet-30? (Epoch, lr, optimizer, batchsize...etc.)

![image](https://user-images.githubusercontent.com/31557552/114716716-476a1f80-9d6f-11eb-88b2-b6b0a1227954.png)

**Comment by jihoontack** [comment]
Hi, thank you for your interest!

Here is the implementation of `sup_ce.py`. You should put it in the `training/sup` folder

The training configuration is as follows:
- Epoch 200
- batch size 128
- weight decay 5e-4
- The learning rate starts at 0.1 and is dropped by a factor of 10 at 100, 150 epoch (i.e., step_decay)
- Optimizer SGD without LARS
- Augmentation: horizontal flipping and random cropping (the basic form)

Thank you

```
import time

import torch.optim

import models.transform_layers as TL
from utils.utils import AverageMeter, normalize

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
HFRC = TL.HorizontalFlipRandomCrop(max_range=0.125).to(device)


def train(P, epoch, model, criterion, optimizer, scheduler, loader, logger=None,
          transform=None, linear=None, linear_optim=None):

    if logger is None:
        log_ = print
    else:
        log_ = logger.log

    batch_time = AverageMeter()
    data_time = AverageMeter()

    losses = dict()
    losses['cls'] = AverageMeter()

    check = time.time()
    for n, (images, labels) in enumerate(loader):
        model.train()
        count = n * P.n_gpus  # number of trained samples

        data_time.update(time.time() - check)
        check = time.time()

        if P.dataset != 'imagenet':
            batch_size = images.size(0)
            images = images.to(device)
            images = HFRC(images)  # transform
        else:
            batch_size = images[0].size(0)
            images = images[0].to(device)

        labels = labels.to(device)

        outputs = model(images)
        loss_ce = criterion(outputs, labels)

        ### total loss ###
        loss = loss_ce

        optimizer.zero_grad()
        loss.backward()
        optimizer.step()

        scheduler.step(epoch - 1 + n / len(loader))
        lr = optimizer.param_groups[0]['lr']

        batch_time.update(time.time() - check)

        ### Log losses ###
        losses['cls'].update(loss_ce.item(), batch_size)

        if count % 50 == 0:
            log_('[Epoch %3d; %3d] [Time %.3f] [Data %.3f] [LR %.5f]\n'
                 '[LossC %f]' %
                 (epoch, count, batch_time.value, data_time.value, lr,
                  losses['cls'].value))

        check = time.time()

    log_('[DONE] [Time %.3f] [Data %.3f] [LossC %f]' %
         (batch_time.average, data_time.average,
          losses['cls'].average))

    if logger is not None:
        logger.scalar_summary('train/loss_cls', losses['cls'].average, epoch)
        logger.scalar_summary('train/batch_time', batch_time.average, epoch)
```

**Comment by wjun0830** [comment]
Great Thanks!

**Comment by wjun0830** [comment]
Hi.

I have tried to reproduce the performance in your paper.
One thing weird is that the AUROC score in Multi-labeled OOD is much better than you reported.
For example, the AUROC score for CIFAR-100 is 91.8 on average and 96.5 for the LSUN dataset.

I also tried CSI and SupCLR+other modules and they were also better than your reported performance.

The only difference when executing is that I used batch size 64 since I only used 2 GPUs.
Are there any possible points that I may be doing something wrong?
